workos/event_objects/directory.py

```diff
@@ -0,0 +1,216 @@
+    state: DirectoryState
+    domains: List[WorkOSOrganizationDomain]
+    organization_id: str
+    created_at: str
+    updated_at: str
+    external_key: str
+    object: Literal["directory"]
+
+    @classmethod
+    def construct_from_response(cls, response: dict):
+        instance = cls()
+        for k, v in response.items():
+            if k == "domains":
+                domains = []
+                for domain in v:
+                    domains.append(
+                        WorkOSOrganizationDomain.construct_from_response(domain)
+                    )
+                setattr(instance, k, domains)
+            elif k == "type":
+                setattr(instance, k, DirectoryType(v))
+            elif k == "state":
+                setattr(instance, k, DirectoryState(v))
+            else:
+                setattr(instance, k, v)
+
+        return instance
+
+    def to_dict(self) -> DirectoryEventWithLegacyFields:
+        return self.__dict__
+
+    def __str__(self):
+        return f"{self.__class__.__name__} {self.to_dict()}"
+
+
+class DirectoryEvent:
+    id: str
+    name: str
+    type: DirectoryType
+    state: DirectoryState
+    organization_id: str
+    created_at: str
+    updated_at: str
+    object: Literal["directory"]
+
+
+class WorkOSDirectoryEvent:
+    id: str
+    name: str
+    type: DirectoryType
+    state: DirectoryState
+    organization_id: str
+    created_at: str
+    updated_at: str
+    object: Literal["directory"]
+
+    @classmethod
+    def construct_from_response(cls, response: dict):
+        instance = cls()
+        for k, v in response.items():
+            if k == "type":
+                setattr(instance, k, DirectoryType(v))
+            elif k == "state":
+                setattr(instance, k, DirectoryState(v))
+            else:
+                setattr(instance, k, v)
+
+        return instance
+
+    def to_dict(self) -> DirectoryEvent:
+        return self.__dict__
+
+    def __str__(self):
+        return f"{self.__class__.__name__} {self.to_dict()}"
+
+
+class DirectoryActivatedEvent(TypedDict):
+    event: str
+    id: str
+    created_at: str
+    data: DirectoryEventWithLegacyFields
+
+
+class WorkOSDirectoryActivatedEvent:
+    event: Literal["dsync.activated"] = "dsync.activated"
+    id: str
+    created_at: str
+    data: WorkOSDirectoryEventWithLegacyFields
+
+    @classmethod
+    def construct_from_response(cls, response: dict):
+        instance = cls()
+        for k, v in response.items():
+            if k == "data":
+                setattr(
+                    instance,
+                    k,
+                    WorkOSDirectoryEventWithLegacyFields.construct_from_response(v),
+                )
+            else:
+                setattr(instance, k, v)
+
+        return instance
+
+    def to_dict(self) -> DirectoryActivatedEvent:
+        return self.__dict__
+
+    def __str__(self):
+        return f"{self.__class__.__name__} {self.to_dict()}"
```

Comment:
Maybe this refactor can come later, but every single Event we define will have...
```python
    event: Literal[EVENT_NAME_LITERAL] = EVENT_NAME_LITERAL
    id: str
    created_at: str
    data: EVENT_DATA_CLASS
```

Can have a single `WorkOSEvent` base class that's parameterized with the two bits of data that actually change? I think the actual implementation of the class looks the same across all events.